### PR TITLE
User/shawkins/colection stats

### PR DESCRIPTION
**Pull request by shawkinsl**
Whew! This was supposed to be a simple feature!

image

So here's everything that's changed:

**Set Collection Progress**

See the screenshot above. Helps you decide if you should keep grinding sealed events or call it quits ;)

**WebWorker in `settingsRenderer`**

Iterating over thousands of cards to count them can cause slight hiccups in the settings window, and this becomes even more pronounced when you do a little extra work (like checking the card's rarity, etc). Solution? Put the work into a WebWorker to unblock the UI "thread" while it's iterating. Unfortunately, it turns out, there's no way to access node modules within webworkers (they're pure js only) => no mtga module in worker contexts :( that means we have to do pretty gross things to access the mtga module's data structures (`allCards.attributes.cards[<grpId>].attributes.prettyName` for example).

**Supporters -> Feedback**

This will allow users to give directly from the app. Talks to the webtask API and (for now) stuffs feedback into a mongo collection.

### CODE REVIEW

File: electron/settings.html [hunk 1]
```diff
@@ -51,9 +51,9 @@ <h5 class="nav-group-title">Other Stuff</h5>
           <span class="icon icon-info"></span>
           About
         </a>
-        <a value="supporters" rv-settingspaneactive="settingsPaneIndex" class="nav-group-item" id="supporters-nav">
+        <a value="feedback" rv-settingspaneactive="settingsPaneIndex" class="nav-group-item" id="feedback-nav">
           <span class="icon icon-heart"></span>
-          Supporters
```
Comment (smkiewel) [inline electron/settings.html]: I'm kinda down on removing the supporter pane entirely. Por que los nos dos?

File: electron/settings.html [hunk 2]
```diff
@@ -290,7 +290,7 @@ <h3>Session Win/Loss Counters</h3>
       <div id="theme-pane" value="theme" rv-showsettingspane="settingsPaneIndex">
         <div class="settings-container">
           <div class="settings-row">
-            <input rv-checked="useFlat" id="enable-flat-theme-toggle" type="checkbox" data-toggle="toggle"> Enable flat theme <span class="helper-text">Changes card display to a simpler "flat" display
+            <input rv-checked="useFlat" id="enable-flat-theme-toggle" type="checkbox" data-toggle="toggle"> Enable flat theme <span class="helper-text">Changes card display to a simpler "flat" display</span>
```
Comment (smkiewel) [inline electron/settings.html]: Thx for the catch.

File: electron/settings.html [hunk 3]
```diff
@@ -732,47 +764,56 @@ <h3>Attributions</h3>
...
-              <li>CommiePuddin <a href="https://www.twitch.tv/commiepuddin"><i class="fa fa-twitch social-icon"></i></a></li>
-              <li>Xanatos</li>
-              <li>Chriss</li>
-            </ul>
-            <p>Support us <a href="https://www.patreon.com/mtgatracker">on Patreon</a> to get your name on the list!</p>
+            <div class="settings-row">
+              <select id="feedback-type-select" class="form-control">
+                <option>Bug Report</option>
+                <option>Feature Request</option>
+                <option>A Love Letter w/ Chocolates</option>
```
Comment (smkiewel) [inline electron/settings.html]: I don't really like chocolate....

File: electron/settingsRenderer.js [hunk 4]
```diff
@@ -250,6 +251,22 @@ rivets.binders.deckid = function(el, value) {
   el.setAttribute('data-deckid', value);
 }
 
+rivets.binders.mythicprogress = function(el, value) {
+  el.style.width = Math.max(10, (100 * value.mythicOwned / value.mythicTotal)) + "%"
```
Comment (smkiewel) [inline electron/settingsRenderer.js]: Why default of 10%?

File: electron/settingsRenderer.js [hunk 5]
```diff
@@ -263,6 +280,101 @@ var firstCounterAdjButtonClicked = true;
 document.addEventListener("DOMContentLoaded", function(event) {
   rivets.bind(document.getElementById('container'), settingsData)
 
+  let collectionWorker = buildWorker(e => {
+    var allCards;
+    var cardSets = {}
+    var playerCardCounts = {}
+    onmessage = event => {
+      if (event.data.allCards) {
```
Comment (smkiewel) [inline electron/settingsRenderer.js]: Why aren't we just calculating this once and storing it?

**Comment by smkiewel** [comment]
Maybe also set name, not just abbreviation?

File: electron/settingsRenderer.js (hunk 4 above)
Comment (shawkinsl) [reply electron/settingsRenderer.js]: If you don't set some mimimum width, then very small percentages will render very poorly:

image

Unfortunately, this has a side effect of misrepresenting 0 collected rows (it looks like you have collected some), but imo the `0 / 123 Mythics` should make that trade-off ok. What do you think?

File: electron/settingsRenderer.js (hunk 5 above)
Comment (shawkinsl) [reply electron/settingsRenderer.js]: That's pretty much what is happening here; the value is only sent to the webworker once. This is def a hack that's explained in the chunk of comments below.

File: electron/settings.html (hunk 2 above)
Comment (shawkinsl) [reply electron/settings.html]: > Maybe also set name, not just abbreviation?

I can do one better: 
image

Turns out these images are in the WotC media kit and should be fair game to use.

File: electron/settingsRenderer.js (hunk 4 above)
Comment (smkiewel) [reply electron/settingsRenderer.js]: Maybe do a little gymnastics to move the "X/Y Rarity" to the other part of the bar if the collected is too small? Maybe shrink the bars, putting the percentage on the collected if large enough and "X/Y" at the end after the bar?

File: electron/settingsRenderer.js (hunk 5 above)
Comment (smkiewel) [reply electron/settingsRenderer.js]: No I mean, why don't we go "Ok there are 64 Mythics in Ixalan, 224 Rares..." in development then store that info in a json object for reading in production. No need to have the webworker ever calculate it.

File: electron/settings.html (hunk 2 above)
Comment (smkiewel) [reply electron/settings.html]: Nice!

Comment (shawkinsl) [reply electron/settings.html]: haha, not sure why this comment ended up here, but :shrug:

File: electron/settingsRenderer.js (hunk 5 above)
Comment (shawkinsl) [reply electron/settingsRenderer.js]: :shrug: it's not a bad idea, but I do like the idea of letting the mtga module be the authoritative source. (And, in this case, even if the mtga module did calculate those numbers, we'd miss out on it in webworkers--that is, unless it's a static number hardcoded in, which I wouldn't be too excited about)

Comment (smkiewel) [reply electron/settingsRenderer.js]: I can see that argument. I can also see the argument, however, that it's much easier to maintain an object that gets updated once every 4 months than it is to maintain some code. Also, less processing generally better. Now I don't feel strongly on this. Just thought I'd offer this opinion on it.

File: electron/settingsRenderer.js (hunk 4 above)
Comment (shawkinsl) [reply electron/settingsRenderer.js]: Good idea. How does this look?

image

(commit inc.)

Comment (smkiewel) [reply electron/settingsRenderer.js]: There we go.

**Comment by smkiewel** [comment]
Hey, I missed a slider in my Settings overhaul, and it just so happens to be on the collections pane, which I basically didn't touch, so I missed it. Fortunately, I'm looking at your collection page and seeing the issue. Would you mind adding class `slidervalue` to line 393?

**Comment by smkiewel** [comment]
And if I were going to pick nits (which I clearly am), `.progress-text` needs `padding-top: 2px`.

**Pull request merged**